tests/ui_tools/test_popups.py

```diff
@@ -1604,6 +1605,52 @@ def test_checkbox_toggle_visual_notification(
         toggle_visual_notify_status.assert_called_once_with(stream_id)
 
 
+class TestTopicInfoView:
+    @pytest.fixture(autouse=True)
+    def mock_external_classes(
+        self, mocker: MockerFixture, general_stream: Dict[str, Any], topics: List[str]
+    ) -> None:
+        self.controller = mocker.Mock()
+
+        mocker.patch.object(
+            self.controller, "maximum_popup_dimensions", return_value=(64, 64)
+        )
+        mocker.patch(LISTWALKER, return_value=[])
+        self.stream_id = general_stream["stream_id"]
+        self.topic = topics[0]
+        self.controller.model.stream_dict = {self.stream_id: general_stream}
+
+        self.topic_info_view = TopicInfoView(
+            self.controller, self.stream_id, self.topic
+        )
+
+    @pytest.mark.parametrize(
+        "topic_name, expected_topic_button_label",
+        [
+            ("hi!", "Resolve Topic"),
+            (f"{RESOLVED_TOPIC_PREFIX}" + "hi!", "Unresolve Topic"),
```

Comment:
Nit: Unless you plan to extract the 'hi!', this can be a combined f-string without the addition.